unstructured/ingest/connector/airtable.py

```diff
@@ -48,39 +50,101 @@ class AirtableIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
     """
 
     connector_config: SimpleAirtableConfig
-    file_meta: AirtableFileMeta
+    table_meta: AirtableTableMeta
     registry_name: str = "airtable"
 
     @property
     def filename(self):
         return (
             Path(self.read_config.download_dir)
-            / self.file_meta.base_id
-            / f"{self.file_meta.table_id}.csv"
+            / self.table_meta.base_id
+            / f"{self.table_meta.table_id}.csv"
         ).resolve()
 
     @property
     def _output_filename(self):
         """Create output file path based on output directory, base id, and table id"""
-        output_file = f"{self.file_meta.table_id}.json"
-        return Path(self.partition_config.output_dir) / self.file_meta.base_id / output_file
+        output_file = f"{self.table_meta.table_id}.json"
+        return Path(self.partition_config.output_dir) / self.table_meta.base_id / output_file
+
+    @property
+    def record_locator(self) -> t.Optional[t.Dict[str, t.Any]]:
+        return {
+            "base_id": self.table_meta.base_id,
+            "table_id": self.table_meta.table_id,
+            "view_id": self.table_meta.view_id,
+        }
+
+    @property
+    def version(self) -> t.Optional[str]:
+        return None
+
+    @requires_dependencies(["pyairtable"], extras="airtable")
+    def _get_table_rows(self):
+        from pyairtable import Api
+
+        api = Api(self.connector_config.personal_access_token)
+        try:
+            table = api.table(self.table_meta.base_id, self.table_meta.table_id)
+            table_url = table.url
+            rows = table.all(
+                view=self.table_meta.view_id,
+            )
+        except Exception as e:
+            # TODO: more specific error handling?
+            logger.error(e)
+            return None, None
```

Comment:
errrm why are we swallowing errors here? seems like if we can't pull the table this _should_ fail, right?

Reply:
per conversation, let's narrow this to only catch the error if it doesn't exist on remote (i.e. not if something like a connection issue occurs)